MNE/disp3D/adapters/ecdview.cpp

```diff
@@ -0,0 +1,152 @@
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief    ECDView class definition.
+*
+*/
+
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "ecdview.h"
+
+#include <inverse/dipoleFit/dipole_fit_settings.h>
+#include <inverse/dipoleFit/ecd_set.h>
+#include <mne/mne_bem.h>
+
+#include <disp3D/view3D.h>
+#include <disp3D/model/data3Dtreemodel.h>
+#include <disp3D/control/control3dwidget.h>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QGridLayout>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace DISP3DLIB;
+using namespace INVERSELIB;
+using namespace MNELIB;
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+ECDView::ECDView(const DipoleFitSettings& dipFitSettings, const ECDSet& ecdSet, QWidget* parent)
+: QWidget(parent)
+, m_p3DView(View3D::SPtr(new View3D()))
+, m_pData3DModel(Data3DTreeModel::SPtr(new Data3DTreeModel()))
+{
+    //Init 3D View
+    m_p3DView->setModel(m_pData3DModel);
+
+    QStringList slControlFlags;
+    slControlFlags << "Data" << "View" << "Light";
+    m_pControl3DView = Control3DWidget::SPtr(new Control3DWidget(this, slControlFlags));
+
+    m_pControl3DView->init(m_pData3DModel, m_p3DView);
+
+    //Read mri transform
+    QFile file(dipFitSettings.mriname);
+    ECDSet ecdSetTrans = ecdSet;
+
+    if(file.exists()) {
```

Comment:
when file not exists, than throw some error/warning message that transoformation might be wrong due to missing mri file